File: cpp/h10.parent/Student.cpp
#include "Student.h"

namespace PersonSpace {
  /******* CONSTRUCTORS *******/
  Student::Student(const char* n)           { setName(n); }
  Student::Student(const char* n, double g) { setName(n); setGPA(g); }

  /******* MUTATORS (SETTERS) *******/
  void   Student::setGPA(double g) { gpa = g; }
  /******* ACESSORS (GETTERS) *******/
  double Student::getGPA() { return gpa; }
}

File: java/h10.parent/Student.java
package person_space;

public class Student extends Person
{
  /******* FIELDS *******/
  private Double gpa;

  /******* CONSTRUCTOR *******/
  public Student(String name)             { this.setName(name); }
  public Student(String name, Double gpa) { this.setName(name); this.setGPA(gpa); }

  /******* MUTATORS (SETTERS) *******/
  public void   setGPA(Double gpa) { this.gpa = gpa; }

  /******* ACCESSORS (GETTERS) *******/
  public Double getGPA()           { return gpa; }
}
